### PR TITLE
chore(world-module-erc20): export erc20 module from internal

### PR DESCRIPTION
**Pull request by vdrg**

### CODE REVIEW

File: .changeset/gold-moles-punch.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+---
+"@latticexyz/world-module-erc20": patch
```
Comment (holic) [inline .changeset/gold-moles-punch.md:2]: since we haven't released this module yet, we can update the previous changeset instead of a new one!

Comment (vdrg) [reply .changeset/gold-moles-punch.md:2]: oh right, makes sense, ty

File: packages/world-module-erc20/ts/defineERC20Module.ts [hunk 2]
```diff
@@ -1,12 +1,13 @@
 import { encodeAbiParameters, stringToHex } from "viem";
+import { Module } from "@latticexyz/world/config/v2";
```
Comment (holic) [inline packages/world-module-erc20/ts/defineERC20Module.ts:2]: we should prob add this as a proper export from world package

File: .changeset/giant-birds-argue.md [hunk 3]
```diff
@@ -7,10 +7,10 @@ The new ERC20 World Module provides a simpler alternative to the ERC20 Puppet Mo
 To install this module, you can import and define the module configuration from the NPM package:
 
 ```typescript
-import { defineERC20Config } from "@latticexyz/world-module-erc20";
+import { defineERC20Module } from "@latticexyz/world-module-erc20/internal";
 
 // Add the output of this function to your World's modules
-const config = defineERC20Config({ namespace: "erc20Namespace", name: "MyToken", symbol: "MTK" });
+const moduleDefinition = defineERC20Module({ namespace: "erc20Namespace", name: "MyToken", symbol: "MTK" });
```
Comment (holic) [inline .changeset/giant-birds-argue.md:13]:
```suggestion
const erc20Module = defineERC20Module({ namespace: "erc20Namespace", name: "MyToken", symbol: "MTK" });
```

**Review by holic — DISMISSED** [review]

File: packages/world-module-erc20/ts/defineERC20Module.ts [hunk 4]
```diff
@@ -1,12 +1,13 @@
...
 
-export type DefineERC20ConfigInput = {
+export type DefineERC20ModuleInput = {
   namespace: string;
   name: string;
   symbol: string;
 };
 
-export function defineERC20Config({ namespace, name, symbol }: DefineERC20ConfigInput) {
+export function defineERC20Module({ namespace, name, symbol }: DefineERC20ModuleInput): Module {
```
Comment (holic) [inline packages/world-module-erc20/ts/defineERC20Module.ts:10]: this should prob use `ModuleInput` as its being used as input for the config 

```suggestion
export function defineERC20Module({ namespace, name, symbol }: DefineERC20ModuleInput): ModuleInput {
```

File: packages/world-module-erc20/package.json [hunk 5]
```diff
@@ -10,17 +10,17 @@
   "license": "MIT",
   "type": "module",
   "exports": {
-    "./mud.config": "./dist/mud.config.js",
-    ".": "./dist/index.js",
-    "./out/*": "./out/*"
+    "internal": "./dist/internal.js",
+    "mud.config": "./dist/mud.config.js",
+    "out/*": "./out/*"
```
Comment (holic) [inline packages/world-module-erc20/package.json:15]: ooc why did these change away from `./` prefix? iirc this had some weird downstream consequences

Comment (vdrg) [reply packages/world-module-erc20/package.json:15]: reverting, I must have gotten confused, thanks for catching that

File: packages/world/ts/exports/internal.ts [hunk 6]
```diff
@@ -11,3 +11,4 @@ export * from "../getFunctions";
 export * from "../getWorldAbi";
 
 export { resolveTableId, resolveWithContext } from "../config/v2/dynamicResolution";
+export { ModuleInput } from "../config/v2/input";
```
Comment (holic) [inline packages/world/ts/exports/internal.ts:14]: fwiw it's safe to just `export *` here since its internal

**Review by holic — APPROVED** [review]

**Pull request merged**